docs/source/guide/upgrading.rst

```diff
@@ -27,3 +27,11 @@ Notes to refer to when upgrading ``boto3`` versions.
     objects for transfers, use the newly added ``upload_fileobj``
     and ``download_fileobj`` methods that support both nonmultipart and
     multipart transfers.
+
+  * By default, all managed transfer methods are now threaded. In prior
```

Comment:
Oh, this isn't quite what I was thinking.  The way you have it now, this is under section 1.4.0, which is wrong.  I was suggesting breaking this out by point version, so people who are upgrading from 1.4.0 (or 1.4.1) up to 1.4.2 have specific upgrade notes they can read.